src/uwtools/drivers/ungrib.py

```diff
@@ -0,0 +1,175 @@
+        path.parent.mkdir(parents=True, exist_ok=True)
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=d,
+            path=path,
+        )
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.gribfile_aaa(),
+            self.namelist_wps(),
+            self.runscript(),
+            self.vtable(),
+        ]
+
+    @tasks
+    def run(self):
+        """
+        A run.
+        """
+        yield self._taskname("run")
+        yield (self._run_via_batch_submission() if self._batch else self._run_via_local_execution())
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        envcmds = self._driver_config.get("execution", {}).get("envcmds", [])
+        execution = [self._runcmd, "test $? -eq 0 && touch %s/done" % self._rundir]
+        scheduler = self._scheduler if self._batch else None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        rs = self._runscript(envcmds=envcmds, execution=execution, scheduler=scheduler)
+        with open(path, "w", encoding="utf-8") as f:
+            print(rs, file=f)
+        os.chmod(path, os.stat(path).st_mode | stat.S_IEXEC)
+
+    @task
+    def vtable(self):
+        """
+        The Vtable.
+        """
+        path = self._rundir / "Vtable"
+        yield self._taskname(str(path))
+        yield asset(path, path.is_symlink)
+        yield None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        path.symlink_to(Path(self._driver_config["vtable"]))
+
+    # Private helper methods
+
+    @property
+    def _driver_config(self) -> Dict[str, Any]:
+        """
+        Returns the config block specific to this driver.
+        """
+        driver_config: Dict[str, Any] = self._config["ungrib"]
+        return driver_config
+
+    @property
+    def _resources(self) -> Dict[str, Any]:
+        """
+        Returns configuration data for the runscript.
+        """
+        return {
+            "account": self._config["platform"]["account"],
+            "rundir": self._rundir,
+            "scheduler": self._config["platform"]["scheduler"],
+            **self._driver_config.get("execution", {}).get("batchargs", {}),
+        }
+
+    @property
+    def _runscript_path(self) -> Path:
+        """
+        Returns the path to the runscript.
+        """
+        return self._rundir / "runscript.ungrib"
+
+    def _taskname(self, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s ungrib %s" % (self._cycle.strftime("%Y%m%d %HZ"), suffix)
```

Comment:
To summarize and extend several previous comments, several of the methods defined here were recently factored out to the `Driver` base class, so they can be removed or simplified here. I think, but please verify:
- `run()` can be deleted, as its's present in the base class.
- `runscript()` can be simplified by calling `self._write_runscript()`.
- `_driver_config()` can be deleted, as it's present in the base class.
- `_driver_name()` should be implemented as a `@property` method here.
- `_runscript_path()` can be deleted, as it's present in the base class.

Reply:
This was my list as well and we're still running successfully on Jet after doing the above.